src/common/SessionConfig.cpp

```
@@ -141,6 +141,8 @@ SessionConfig::SessionConfig(const Config & conf) {
     };
     ConfigDefault<std::string> strValues [] = {
         {&defaultUri, "dfs.default.uri", "hdfs://localhost:9000" },
+        {&defaultFS, "fs.defaultFS", "hdfs://localhost:8020" },
+        {&defaultName, "fs.default.name", "hdfs://localhost:8020" },
```

What I am seeing when I change this code the way I think you are suggesting is that the configured value for one of these settings seems to get overwritten by the hard-coded default values for the others which results in function test failures.  Here is what I changed that got that result:
```
          {&defaultUri, "dfs.default.uri", "hdfs://localhost:9000" },
          {&defaultFS, "fs.defaultFS", "hdfs://localhost:8020" },
          {&defaultName, "fs.default.name", "hdfs://localhost:8020" },
```
becomes:

```
          {&defaultFS, "dfs.default.uri", "hdfs://localhost:9000" },
          {&defaultFS, "fs.defaultFS", "hdfs://localhost:9000" },
          {&defaultFS, "fs.default.name", "hdfs://localhost:9000" },
```
with appropriate changes to the access methods and removal of the stray members (not shown here).  Thus making all three settings synonyms for one config variable.

Did you have a different change in mind?